From: Ben Hutchings <ben@decadent.org.uk>
Date: Mon, 29 Dec 2014 00:59:35 +0100
Subject: PCI: Avoid ABI change in 3.16.7-ckt3
Forwarded: not-needed

Move the added bitfield pci_dev::no_64bit_msi to the end of its
bitfield group and hide it from genksyms.

--- a/include/linux/pci.h
+++ b/include/linux/pci.h
@@ -331,7 +331,6 @@ struct pci_dev {
 	unsigned int	is_added:1;
 	unsigned int	is_busmaster:1; /* device is busmaster */
 	unsigned int	no_msi:1;	/* device may not use msi */
-	unsigned int	no_64bit_msi:1; /* device may only use 32-bit MSIs */
 	unsigned int	block_cfg_access:1;	/* config space access is blocked */
 	unsigned int	broken_parity_status:1;	/* Device generates false positive parity */
 	unsigned int	irq_reroute_variant:2;	/* device needs IRQ rerouting variant */
@@ -349,6 +348,10 @@ struct pci_dev {
 	unsigned int	__aer_firmware_first:1;
 	unsigned int	broken_intx_masking:1;
 	unsigned int	io_window_1k:1;	/* Intel P2P bridge 1K I/O windows */
+#ifndef __GENKSYMS__
+	unsigned int	no_64bit_msi:1; /* device may only use 32-bit MSIs */
+	/* 8 bits spare */
+#endif
 	pci_dev_flags_t dev_flags;
 	atomic_t	enable_cnt;	/* pci_enable_device has been called */
 
